src/segmenter.rs

```diff
@@ -49,10 +50,149 @@ pub fn split_sentences_with_python_de(text: &str) -> Vec<String> {
+            "hurt",  #
+            "mech",  #
+            "wyj",  # wyj
+            "pt",  # pod tytułem
+            "zew",  # zewnętrzny
+            # "Sp",
+        ]
+
+        position_abbrev = [
+            "Ks", "Abp", "abp", "bp", "dr", "kard", "mgr", "prof", "zwycz", "hab", "arch",
+            "arch.kraj", "B.Sc", "Ph.D", "lek", "med", "n.med", "bł", "św", "hr", "dziek", "ks", "inz",
+        ]
+
+        roman_abbrev = (
+            []
+        )
+
+        quantity_abbrev = [
+            "mln", "tys", "obr./min", "km/godz", "godz", "egz", "ha", "j.m", "cal", "obj",
+            "alk", "wag", "op", "wk", "mm",
+            "MB",  # mega bajty
+            "Mb",  # mega bity
+            "jedn",  # jednostkowe
+            "obr",  # obroty
+            "szt",  # sztuk
+        ]
+
+        actions_abbrev = [
+            "tłum", "tlum", "zob", "wym", "w/wym", "pot", "ww", "ogł", "wyd", "min",
+            "m.i", "m.in", "in", "im", "muz", "tj", "dot", "wsp", "właść", "właśc", "przedr",
+            "czyt", "proj", "dosł", "hist", "daw", "zwł", "zaw", "późn", "spr", "jw",
+            "odp",  # odpowiedź
+            "symb",  # symbol
+            "klaw",  # klawiaturowe
+        ]
+
+        place_abbrev = [
+            "śl",
+            "płd",
+            "geogr",
+            "zs",
+            "pom",  # pomorskie
+            "kuj-pom",  # kujawsko pomorskie
+        ]
+
+        lang_abbrev = [
+            "jęz", "fr", "franc", "ukr", "ang", "gr", "hebr", "czes", "pol", "niem",
+            "arab", "egip", "hiszp", "jap", "chin", "kor", "tyb", "wiet", "sum", "chor",
+            "słow", "węg", "ros", "boś", "szw",
+        ]
+
+        administration = [
+            "dz.urz",  # dziennik urzędowy
+            "póź.zm",
+            "rej",  # rejestr, rejestracyjny dowód
+            "sygn",  # sygnatura
+            "Dz.U",  # dziennik ustaw
+            "woj",  # województow
+            "ozn",  #
+            "ust",  # ustawa
+            "ref",  # ref
+            "dz",
+            "akt",  # akta
+        ]
+
+        time = [
+            "tyg",  # tygodniu
+        ]
+
+        military_abbrev = [
+            "kpt", "kpr", "obs", "pil", "mjr", "płk", "dypl", "pp", "gw", "dyw",
+            "ppłk", "mar", "marsz", "rez", "ppor", "DPanc", "BPanc", "DKaw", "p.uł",
+            "sierż", "post", "asp", "podinsp", "nadkom", "rtm",
+            "bryg",  # brygady
+            "szt",  # sztabowy
+            "kom",  # komendant, tel. komórka
+        ]
+
+        extra_abbreviations = (
+            extra_abbreviations
+            + position_abbrev
+            + quantity_abbrev
+            + place_abbrev
+            + actions_abbrev
+            + lang_abbrev
+            + administration
+            + time
+            + military_abbrev
+        )
+
+        sentence_tokenizer = nltk.data.load("tokenizers/punkt/polish.pickle")
+        sentence_tokenizer._params.abbrev_types.update(extra_abbreviations)
```

Comment:
These abbreviations would then still end up in the sentences right? I see that a few of those are indeed being replaced or checked for, but quite a few of those are not. Are those all easily pronounceable? Alternatively I would suggest to add those to `replacements` as well and then we could also simplify this part here. What do you think?

Reply:
Those abbreviations do end up in the split sentences. The reason for them being fed to the python punkt segmenter is that without them some sentences get split incorrectly (i.e. "Jego współpracownikami byli inż. A. Schmidta oraz technolodzy: Lipko, Obryś i Rohn." is split into two). So here it is purely to improve general quality of the splitting. They are latter filtered out by the abbreviations regex if not replaced as most of may be read incorrectly. 
Sadly replacements are not always possible as many abbreviations in expanded form change conjugation or vary in hard to manage way. But it came to my head that maybe this python part can be moved to separate .py file to hide the not-so-pretty word listings and in segmenter.rs we can just call one func :)

Reply:
So all of these here have to be followed by a period then? So that's kinda implicit when defining abbreviations here? If so, I'm totally fine with this and all is good :)

> But it came to my head that maybe this python part can be moved to separate .py file to hide the not-so-pretty word listings and in segmenter.rs we can just call one func :)

When I set it up I couldn't figure out how to do that, but I didn't want to spend too much time on it. If such a setup is doable, I'd definitely prefer that as well. However I don't see this as a requirement to get this PR merged.

Reply:
Yes, most of polish abbreviations are ended with a dot and even some exceptions which should not have dot are found to have one :)
As for cleaning up python code into other file... it seems more difficult than I expected. I don't quite grasp how rust works and inline python is not an exception. It seems to me that regular .py files are lacking something for inline python to process them correctly. But I found some workaround. I can have a separate rust function which creates context with all the arrays of strings and return the context which is then used to run the splitting. The context setup function can be a separate file hiding what does not have to be visible in the segmenter.rs. Anyway I need to learn rust :D

Reply:
> I can have a separate rust function which creates context with all the arrays of strings and return the context which is then used to run the splitting. The context setup function can be a separate file hiding what does not have to be visible in the segmenter.rs.

I'm not sure I understand your workaround. In any case, I think your additions are fine for now, I wouldn't block on that :)